applications/nrf_desktop/doc/config_channel.rst

```diff
@@ -61,37 +98,255 @@ Setting a configuration value of a device, forwarded by the dongle to a paired d
+  The host tool can access data returned by the response.
+  The host can send a new request.
+* ``CONFIG_STATUS_TIMEOUT`` - The operation timed out on the device.
+  The request was accepted by an application module but the response was not prepared in time.
+  The host can send a new request.
+* ``CONFIG_STATUS_REJECT`` - The operation was rejected.
+  The host can send a new request.
+* ``CONFIG_STATUS_WRITE_FAIL`` - Forwarding configuration channel transaction failed.
+  The host can send a new request.
+* ``CONFIG_STATUS_DISCONNECTED`` - The operation failed because a module or a device addressed by the request did not respond.
+  This can happen when Bluetooth® LE Peripheral disconnects.
+  The host can send a new request.
+
+Data length
+-----------
+
+Indicates how many meaningful bytes of data the request or the response holds.
+As the HID feature report has a fixed length, this field indicates the size of the meaningful data.
+
+Data
+----
+
+The piece of data of the length defined by Data length, related to the request or the response.
+
+.. _nrf_desktop_config_channel_device_discovery:
+
+Device discovery
+================
+
+Before the device can be configured over the configuration channel, it needs to be discovered by the host scripts.
+The discovery procedure identifies the device and gets information about its configurable application modules and their options.
+
+If you develop a custom host tool, you need to ensure your tool performs the following steps to discover the device:
+
+1. Enumerate all of the connected HID devices.
+#. Filter the enumerated devices using Vendor ID and Product ID specified for your devices.
+#. For each applicable device, perform recipient discovery.
+#. For each device, perform device identification.
+#. For each device, perform module discovery.
+
+.. tip::
+   The host tools do not need to perform complete discovery of the device every time the device is connected.
+   The Module ID and Options ID cannot change until firmware is updated on the device.
+   The module descriptor can be cached by the host for a given device and firmware version.
+
+.. _nrf_desktop_config_channel_recipient_discovery:
+
+Recipient discovery
+-------------------
+
+The Recipient ID equal to ``0`` is used to communicate with any device directly connected to the host.
+
+The configuration channel allows a dongle type device to act as a proxy for Bluetooth® LE Peripheral devices.
+If you develop a custom host tool, you need to ensure your tool performs the following steps to discover all of the peripherals that are connected to the dongle over Bluetooth® LE:
+
+1. Trigger a Recipient ID re-evaluation with the ``CONFIG_STATUS_INDEX_PEERS`` request.
+#. Obtain a list of connected devices by calling the ``CONFIG_STATUS_GET_PEER`` request.
+#. Perform :ref:`nrf_desktop_config_channel_device_identification` and :ref:`nrf_desktop_config_channel_module_discovery` procedures on selected or all connected devices.
```

Comment:
It depends what you understand under term `recipient discovery`. If you wish, I could remove it, but I think that this step could be treated as part of the discovery of Bluetooth connected peripherals, because it's actually needed to configure the peripherals over configuration channel. This approach in the description is more straightforward.

Reply:
By `recipient discovery` I mean only the part where you get Recipient IDs.
Once you have the list of all devices (local or remote) you can query them for modules&options. But this is a separate step. It is hinted at lines 266-268 and fully described later when we talk about module discovery.
This this step does not belong here.